BLSQ/iaso
# POLIO-1462: stock mgt: connect modals to api

Connect form a, destruction and incident reports to API

Related JIRA tickets : POLIO-1462

## Self proofreading checklist


## Doc
NA (yet)

## Changes

- update keys to match API in modals, tables, validation and translations
- remove mock data
- update validation to make fields required
- mark fields as required in interface
- show doses numbers in Unusable table
- add text placeholder on empty table rows

## How to test

Go to Polio > vaccine module > stock management:
- check data in usable/unusable
- go to stock variation
- for each tab, try to add and edit one or several items
- go back to details view: the changes should be reflected

## Print screen / video

Uploading Screen Recording 2024-02-01 at 17.39.32.mov…




## Notes

- Merges in POLIO-1449-vaccine-stock-management-backend-form-a-destruction-incident-api
- Non blocking changes will be put in separate ticket
- failing python tests will be fixed in base branch
